src/utils/utils.ts

```diff
@@ -86,3 +86,20 @@ export function bufToBn(buf: Buffer) {
 }
 
 export const isBrowser = new Function('try {return this===window;}catch(e){ return false;}');
+
+export async function getJsonResponse<T>(response: Promise<Response>): Promise<T> {
+  let r;
+  try {
+    r = await response;
+  } catch (e) {
+    throw new Error(`Error while communicating with gateway: ${JSON.stringify(e)}`);
+  }
+
+  if (!r?.ok) {
+    const text = await r.text();
```

Comment:
works quite ok for errors thrown through the koa context like this one:
`ctx.throw(400, 'Error message');`

the error thrown by sdk:
`Error: 400: {"message":{"message":"Contract data item binary is not valid."}}`

for the errors thrown through regular Error object:
`throw new Error('Error text.')`

the error thrown by sdk:
`Error: 500: {"message":{}}`

I'm not sure if we could always pass error message in the ctx.body when error is thrown? like this:
`ctx.body = e.message`

Reply:
the last case is be fixed by the changes in the GW - https://github.com/warp-contracts/gateway/pull/161